storage/host-flash/host-flash.c

```diff
@@ -51,28 +52,39 @@ ssize_t hostflash_read(unsigned int addr, void *buff, size_t bufflen)
 
 ssize_t hostflash_write(unsigned int addr, void *buff, size_t bufflen)
 {
-    char currentByte = 0;
-    size_t wrote = 0;
-    ssize_t stat;
+    char tempTab[256];
+    size_t wrote = 0, i;
+    int len;
+    ssize_t stat, readed;
 
     if (addr + bufflen > hostflash_common.flashsz)
         return -EINVAL;
 
     while (wrote != bufflen) {
-        stat = hostflash_read(addr + wrote, &currentByte, 1);
-        if (stat <= 0)
-            return stat;
-
-        currentByte = currentByte & *((char *)buff + wrote);
-        stat = pwrite(hostflash_common.filefd, &currentByte, 1, addr + wrote);
-        if (stat < 0) {
-            if (errno == -EINTR)
-                continue;
-
-            return stat;
+        if ((bufflen - wrote) / sizeof(tempTab))
+            len = sizeof(tempTab);
+        else
+            len = (bufflen - wrote) % sizeof(tempTab);
+
+        readed = hostflash_read(addr + wrote, tempTab, len);
+        if (readed <= 0)
+            return readed;
+
+        for (i = 0; i < readed; i++)
+            tempTab[i] = tempTab[i] & *((char *)buff + wrote + i);
+
+        len = 0;
+        while (len != readed) {
+            stat = pwrite(hostflash_common.filefd, &tempTab, readed - len, addr + wrote + len);
+            if (stat < 0) {
+                if (errno == -EINTR)
```

Comment:
```suggestion
                if (errno == EINTR)
```
`errno` is defined by the ISO C standard, valid error numbers are all positive numbers.

Reply:
Done